File: api/src/main/java/de/storyteller/api/model/Comment.java
package de.storyteller.api.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * Model class for a comment
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Comment {
    private String content;
    private String authorId;
    private String authorName;
    private String chapterId;
    private String blockId;
    private LocalDateTime createdAt;
}

File: webapp/storyteller/src/app/model/comment.ts
/**
 * Comment model class
 */

export interface Comment {
  chapterId: string;
  authorId: string;
  authorName: string;
  content: string;
  blockId: string;
  createdAt: Date;
}

export interface AddCommentRequest {
  content: string;
  chapterId: string;
  blockId: string;
}